examples/synthetics_example_test.go

```diff
@@ -392,3 +398,375 @@ func makeExampleTest() *synthetics.V202101beta1Test {
+	} else {
+		agent.Family = syntheticspb.IPFamily_IP_FAMILY_V6
+	}
+	agent.Name = ""
+	patchReqPayload := &syntheticspb.PatchAgentRequest{
+		Agent: agent,
+		Mask:  &fieldmaskpb.FieldMask{Paths: []string{"agent.family"}},
+	}
+	patchResp, err := client.SyntheticsAdmin.PatchAgent(ctx, patchReqPayload)
+	if err != nil {
+		return err
+	}
+	PrettyPrint(patchResp.GetAgent())
+	fmt.Println()
+
+	// NOTE: as we can't create agents through the API - let's not delete them
+	//fmt.Println("### DELETE AGENT")
+	//deleteReqPayLoad := &syntheticspb.DeleteAgentRequest{Id: agentID}
+	//_, err = client.SyntheticsAdmin.DeleteAgent(ctx, deleteReqPayLoad)
+	//if err != nil {
+	//	return err
+	//}
+	//fmt.Println("Success")
+	//fmt.Println()
+
+	return nil
+}
+
+func runGRPCListAgents(ctx context.Context, client *kentikapi.Client) error {
+	fmt.Println("### LIST AGENTS")
+
+	getAllReq := &syntheticspb.ListAgentsRequest{}
+	getAllResp, err := client.SyntheticsAdmin.ListAgents(ctx, getAllReq)
+	if err != nil {
+		return err
+	}
+
+	if getAllResp.Agents != nil {
+		agents := getAllResp.GetAgents()
+		fmt.Println("Num agents:", len(agents))
+		if getAllResp.InvalidAgentsCount != 0 {
+			fmt.Println("Num invalid agents:", getAllResp.InvalidAgentsCount)
+		}
+		PrettyPrint(agents)
+	} else {
+		fmt.Println("[no agents received]")
+	}
+	fmt.Println()
+
+	return nil
+}
+
+func makeGRPCExampleTest() *syntheticspb.Test {
+	healthSettings := &syntheticspb.HealthSettings{
+		LatencyCritical:     0,
+		LatencyWarning:      0,
+		PacketLossCritical:  0,
+		PacketLossWarning:   0,
+		JitterCritical:      0,
+		JitterWarning:       0,
+		HttpLatencyCritical: 0,
+		HttpLatencyWarning:  0,
+		HttpValidCodes:      []uint32{},
+		DnsValidCodes:       []uint32{},
+	}
+
+	monitorSettings := &syntheticspb.TestMonitoringSettings{
+		ActivationGracePeriod: "2",
+		ActivationTimeUnit:    "m",
+		ActivationTimeWindow:  "5",
+		ActivationTimes:       "3",
+		NotificationChannels:  []string{},
+	}
+
+	ping := &syntheticspb.TestPingSettings{
+		Period: 60,
+		Count:  5,
+		Expiry: 3000,
+	}
+
+	trace := &syntheticspb.TestTraceSettings{
+		Period:   60,
+		Count:    3,
+		Protocol: "udp",
+		Port:     33434,
+		Expiry:   22500,
+		Limit:    30,
+	}
+
+	settings := &syntheticspb.TestSettings{
+		Definition: &syntheticspb.TestSettings_Hostname{
+			Hostname: &syntheticspb.HostnameTest{Target: "dummy-ht.com"},
+		},
+		AgentIds: []string{
+			"890",
+		},
+		Period: 0,
+		Count:  0,
+		Expiry: 0,
+		Limit:  0,
+		Tasks: []string{
+			"ping",
+			"traceroute",
+		},
+		HealthSettings:     healthSettings,
+		MonitoringSettings: monitorSettings,
+		Ping:               ping,
+		Trace:              trace,
+		Port:               443,
+		Protocol:           "icmp",
+		Family:             syntheticspb.IPFamily_IP_FAMILY_DUAL,
+		Servers:            []string{},
+		UseLocalIp:         false,
+		Reciprocal:         false,
+		RollupLevel:        1,
+		Http:               nil,
```

Comment:
`hostname` type test will never have any `http` section. In general, each test type except `application_mesh` and `bgp_monitor` has a section `settings` named after the test type (i.e. for `hostname` test, the name is `hostname`). This section contains test type specific attributes, most importantly the `target` (or `targets`) and eventually more. There are some anomalies (e.g. `dns` and `dns_grid` tests require list of `servers` and it is not in the test type specific section, rather directly in `settings`). Most (if not all) of those anomalies are going to be straightened in `beta2` version of the API (which already exists, but has not been made public yet). This is though just a stepping stone to "stable" version which will have additional schema changes.